jOOQ/src/main/java/org/jooq/impl/DefaultRecordMapper.java

```diff
@@ -343,6 +349,33 @@ private final void init(E instance) {
             return;
         }
 
+        // [#7324] Map Kotlin data classes
+        KClass<? extends E> kotlinClass = JvmClassMappingKt.getKotlinClass(type);
+        if (kotlinClass.isData()) {
+            KFunction<? extends E> primaryConstructor = KClasses.getPrimaryConstructor(kotlinClass);
+            List<KParameter> parameters = primaryConstructor.getParameters();
+
+            if (parameters.size() != fields.length) {
+                throw new MappingException("Primary constructor of Kotlin data class " + type + " did not match row type " + rowType);
+            }
+
+            for (String fieldName : fieldNameStrings(fields)) {
+                String name = StringUtils.toCamelCaseLC(fieldName);
+                boolean found = false;
+                for (KParameter kParameter : parameters) {
+                    String parameterName = kParameter.getName();
+                    if (name.equals(parameterName)) {
+                        found = true;
+                    }
+                }
+                if (!found) {
+                    throw new MappingException("Primary constructor of Kotlin data class " + type + " did not match row type " + rowType);
+                }
```

Comment:
Again, we don't require an exact match between source and target fields. If we don't find a specific field in the target type, then we simply won't map it.

There are pros and cons to each approach, but for consistency reasons with existing `DefaultRecordMapper` behaviour, this exception shouldn't be thrown.